experiment/contractNipopow.sol

```diff
@@ -322,23 +339,23 @@ contract Crosschain {
     return false;
   }
 
-  function finalize_event(bytes32[4] block_of_interest) public returns(bool) {
+  function finalize_event(bytes32[4] memory block_of_interest) public returns(bool) {
     bytes32 hashed_block = hash_header(block_of_interest);
 
     if (events[hashed_block].expire == 0 ||
       block.number < events[hashed_block].expire) {
       return false;
     }
     finalized_events[hashed_block] = true;
-    address author = events[hashed_block].author;
+    address payable author = address(uint160((events[hashed_block].author)));
```

Comment:
can you recall why this was necessary? event.author already looks to be address type

Reply:
Because it is not declare payable and author.transfer is called before later.

Reply:
yep so let's do that